esp/services/ws_access/ws_accessService.cpp

```diff
@@ -2087,24 +2089,36 @@ bool Cws_accessEx::onResourceDelete(IEspContext &context, IEspResourceDeleteRequ
                 if(prefix && *prefix)
                     namebuf.insert(0, prefix);
 
-                secmgr->deleteResource(rtype, namebuf.str(), basednReq->getBasedn(), context.querySecureContext());
+                try
+                {
+                    secmgr->deleteResource(rtype, namebuf.str(), basednReq->getBasedn(), context.querySecureContext());
+                }
+                catch(IException* e)
+                {
+                    StringBuffer emsg;
+                    e->errorMessage(emsg);
+                    errMsgs.appendf("\n%d %s", e->errorCode(), emsg.str());//accumulate error messages/return codes
+                    iLastRC = e->errorCode();
+                    e->Release();
+                }
             }
         }
-        catch(IException* e)
-        {
-            StringBuffer emsg;
-            e->errorMessage(emsg);
-            resp.setRetcode(e->errorCode());
-            resp.setRetmsg(emsg.str());
-            return false;
-        }
+
         catch(...)
         {
             resp.setRetcode(-1);
             resp.setRetmsg("Unknown error");
             return false;
         }
 
+        if (!errMsgs.isEmpty())
+        {
+            resp.setRetcode(iLastRC);
+            errMsgs.append("\n");
```

Comment:
@afishbeck  Is this the correct way to return an exception?  From what I can see the callers catch exceptions that are thrown, and then serialize it in different ways depending on the request format.

Reply:
@afishbeck @ghalliday The reason it was done this way is that I don't want to abort the operation if any single "delete" fails, instead I want to proceed with the remaining ones and report back all the ones that failed

Reply:
yes, that makes sense.  The question was more whether this code here, once the failures have been collected should be thrown as an exception.  (It is a question from ignorance since I don't know what the general pattern is for these services.)

Reply:
@ghalliday @afishbeck It used to throw, so I took your recommendation and, if there are errors,  throw an exception after all deletes are attempted

Reply:
@RussWhitehead @ghalliday sorry it just occurred to me that the exception being thrown when there are multiple is an IMultiException.  In the same place where you catch the exception and append to your error string you should be appending to an IMultiException.  Then throwing that at the end.

The exceptions returned in the response is an array.

Reply:
Here's a fairly simple example:

https://github.com/hpcc-systems/HPCC-Platform/blob/master/esp/services/ws_workunits/ws_workunitsService.cpp#L882